### ISSUE TITLE
[Desktop] - Splash screen or background color customization

### ISSUE THREAD

**Issue by bleroux** [flutter/flutter#41980]
## Use case

For desktop support, it would be nice to have an easy way to configure a splash screen or a window background color (visible before Flutter first frame is rendered).

## Proposal

Add properties in c++ code (win32_flutter_window.cc, flutter_glfw.cc, etc) or in a configuration file   to make this customizable on desktop launchers.

Launching Flutter on desktop is in early stage and is moving fast so feel free to close this feature request if it is too soon to ask for this.

**Comment by cbracken** [comment]
Deduped with https://github.com/flutter/flutter/issues/82538 filed by @jonbhanson.

Documentation will need to be added once we've decided on the approach to take with this:
Page URL: https://flutter.dev/docs/development/ui/advanced/splash-screen.html
Page source: https://github.com/flutter/website/tree/master/src/docs/development/ui/advanced/splash-screen.md

**Comment by loic-sharma** [comment]
Hello, the app's window is now hidden until Flutter has started and drawn the first frame:

https://github.com/flutter/flutter/blob/c60cf97c1904883accf43ad844d7ff11cf446846/examples/hello_world/windows/runner/flutter_window.cpp#L34-L36

In the future, we will build on this to add splash screen functionality to Windows.

**Comment by jonbhanson** [comment]
@loic-sharma will the splash screen functionality in Windows have any type of native implementation that will display before Flutter has drawn the first frame?  This is how my package [flutter_native_splash](https://pub.dev/packages/flutter_native_splash) works on Android, iOS, and web: the splash screen is displayed natively so that it appears immediately.  Then, when the app is ready, the package removes the native splash with `widgetsBinding.allowFirstFrame()`.

**Comment by loic-sharma** [comment]
Yup, we'll likely do something like you suggested. For now, Flutter will hide the app's window until it's ready. We think that's sufficient for desktop apps that start quickly.

**Comment by annd22** [comment]
but in the case I hide the windows then show it programmatically, this white screen still appears in a few sec

**Comment by itsyosefali** [comment]
any updates ?

**Comment by JaviBonilla** [comment]
> Hello, the Windows app's window is now hidden until Flutter has started and drawn the first frame:
> 
> https://github.com/flutter/flutter/blob/c60cf97c1904883accf43ad844d7ff11cf446846/examples/hello_world/windows/runner/flutter_window.cpp#L34-L36
> 
> In the future, we will:
> 
> 1. Update macOS and Linux as well to hide the app until the first frame is drawn
> 2. Add support for splash screen functionality

Any updates about Linux and macOS? Thanks.

**Comment by mstfkhazaal** [comment]
any update?

**Comment by shubhamsinghmutualmobile** [comment]
Hi.. do we have any updates on this ticket?
